### ISSUE TITLE
[$250] Conversation - Error message happen when trying to close an account

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50646]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.48-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5079399
**Email or phone of affected tester (no customers):** testpayment935@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Prerequisites: have 3 accounts you can access
1. Navigate to https://staging.new.expensify.com/
2. Log in and create a group with 2 other accounts you have access
3. Log in to any of the 2 accounts and close it via Settings > Security

## Expected Result:
The account closed without any issue

## Actual Result:
Error message happen when trying to close an account

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

![149999999999999](https://github.com/user-attachments/assets/307d8511-cacb-4653-85d9-a040b88af099)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021846310011679961499</li>
        <li>Upwork Job ID: 1846310011679961499</li>
        <li>Last Price Increase: 2024-10-15</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 104528450</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @rojiphil</details>

**Comment by lanitochka17** [comment]
@johncschuster  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by JKobrynski** [comment]
Hi, I'm Julian from Callstack - expert agency - and I would like to work on this issue.

**Comment by kubabutkiewicz** [comment]
Hello, @JKobrynski is on sick leave unfortunately, I will take over from them

**Comment by kubabutkiewicz** [comment]
Hello @lanitochka17, could you explain more this point `and access the 2 account details page via the header` I am not quite sure what do you mean here?

**Comment by kubabutkiewicz** [comment]
Hey I was trying to reproduce that issue and only time when I go error during closing account was that one below

<img width="731" alt="image" src="https://github.com/user-attachments/assets/553aab95-93ea-4b9c-95b8-db79d7eb05a6">

but I would like to get thoughts someone from backend team because what I see from the code we are displaying errors based on the message which we have in Onyx and them are coming from backend.

**Comment by kubabutkiewicz** [comment]
Adding responses in text form so it will be easier to copy

1st CloseAccount attempt 
```
{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Error\\User\\Undeletable",
    "UUID": "c65019fc-c2d8-4529-8382-045c9c13d6cf",
    "message": "An unknown error has occurred and a technician has been notified. If the issue persists, please message concierge.",
    "title": "Unable to close your account",
    "data": {
        "message": "405 Reports are processing",
        "reimburseDate": null,
        "onyxData": [
            {
                "onyxMethod": "merge",
                "key": "closeAccount",
                "value": {
                    "errors": {
                        "1729753590290406": "An unknown error has occurred and a technician has been notified. If the issue persists, please message concierge."
                    }
                }
            }
        ]
    },
    "htmlMessage": "",
    "onyxData": [
        {
            "onyxMethod": "merge",
            "key": "closeAccount",
            "value": {
                "errors": {
                    "1729753590290406": "An unknown error has occurred and a technician has been notified. If the issue persists, please message concierge."
                }
            }
        }
    ],
    "requestID": "8d781fde8d14bbdb-WAW"
}
```

2nd CloseAccount attempt 
```
{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Libs\\Error\\ExpError",
    "UUID": "d35ea641-3250-4eec-acbd-57058b99cf8b",
    "message": "Unable to retract Chat report",
    "title": "Action not permitted",
    "data": {
        "onyxData": [
            {
                "onyxMethod": "merge",
                "key": "closeAccount",
                "value": {
                    "errors": {
                        "1729754101448227": "Unable to retract Chat report"
                    }
                }
            }
        ]
    },
    "htmlMessage": "",
    "onyxData": [
        {
            "onyxMethod": "merge",
            "key": "closeAccount",
            "value": {
                "errors": {
                    "1729754101448227": "Unable to retract Chat report"
                }
            }
        }
    ],
    "requestID": "8d782c5cbee1b629-WAW"
}
```

**Comment by rojiphil** [comment]
> error is coming from an API so I would like to someone from backend to take a look why its happening

That makes sense. Thanks. As the error is from BE, let's bring in an internal engineer to have a look.
🎀👀🎀 C+ reviewed

**Comment by lanitochka17** [comment]
@kubabutkiewicz The third step is optional. Sorry for the confusion.

**Comment by lanitochka17** [comment]
but issue is still reproducible 

https://github.com/user-attachments/assets/6fc5159c-6b89-41ff-8483-a026113cf40b

**Comment by Beamanator** [comment]
Def looks like an internal issue 👍 

@kubabutkiewicz it sounds like you didn't need many steps to be able to reproduce, you just created an account & then turned around and tried to close it?

**Comment by kubabutkiewicz** [comment]
@Beamanator I followed reproduction steps in issue description and after several tries I was able to reproduce it

**Comment by rojiphil** [comment]
> Def looks like an internal issue 👍

I am unassigning myself as this is an internal issue.  I will keep myself subscribed though.

**Comment by johncschuster** [comment]
@Beamanator looks like this is to you and @JKobrynski! What's the status on this one?

**Comment by johncschuster** [comment]
@Beamanator / @JKobrynski bump!

**Comment by johncschuster** [comment]
Bumping for Melvin. @Beamanator, any updates on this one?

**Comment by johncschuster** [comment]
@Beamanator bump!

**Comment by johncschuster** [comment]
Bump! Any updates, @Beamanator?

**Comment by Beamanator** [comment]
Sorryyyy this says LOW priority so I haven't prioritized it yet 🤣

**Comment by johncschuster** [comment]
Haha I hear ya! In that case, it may be a good idea to decrease the label to match the priority level.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by johncschuster** [comment]
Seems like this might be resolved. We good to close this one, @Beamanator?

**Comment by Beamanator** [comment]
Nice! Yes I agree let's close this out 👍

**Comment by lanitochka17** [comment]
Issue is still reproducible on the latest build 9.0.76-6
The account that triggered this issue is: hywnnoj+0@gmail.com

https://github.com/user-attachments/assets/f22e27c0-9f5e-4058-8a9b-698ba1ae7a1e

**Comment by Beamanator** [comment]
haven't had time yettt but will look next week! 🙏

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by johncschuster** [comment]
Hmm... I'm a bit confused here. @mvtglobally cannot reproduce the issue, but @lanitochka17 can. @lanitochka17, are you able to reproduce this reliably? If so, can you please clarify the reproduction steps?

**Comment by lanitochka17** [comment]
@johncschuster The tester provided accounts where this is reproduced:

[gloginme3+1010sad@gmail.com](mailto:gloginme3+1010sad@gmail.com)
[applausetester+bp0813ap@applause.expensifail.com](mailto:applausetester+bp0813ap@applause.expensifail.com)
[applausetester+bp307two@applause.expensifail.com](mailto:applausetester+bp307two@applause.expensifail.com)

**Comment by mvtglobally** [comment]
@johncschuster  if the issue is intermittent or account specific, another tester running same steps might not be able to reproduce. 
Issue not reproducible during KI retests. (Second week)

**Comment by johncschuster** [comment]
Thanks for those details, @lanitochka17 and @mvtglobally! Sounds like this issue persists!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by Beamanator** [comment]
Couldn't get any logs of `CloseAccount` from the accounts provided, please if this is reproduce again, let me know the email address of the account :D

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Forth week) Good to close?

**Comment by Beamanator** [comment]
Sounds good to me to close this out!
